htcosmos/make_parquet.py

```diff
@@ -94,73 +138,87 @@ def process_pages(filename, page_info_dir, out_dir, meta, limit, model, model_co
-        padded_img = pad_image(img)
-        tlog(f'{page_name} padded to {padded_img.size}')
-        # the padding operation is
-        #want_image_size = 1920
-        #d_w = want_image_size - w
-        #d_h = want_image_size - h
-        #padding = (0,0,d_w, d_h)
-        #padded_img = ImageOps.expand(img, padding, fill="#fff")
-
-        # the aggregation code needs access to the padded image
+        model_id = '0'
+        obj['id'] = model_id
+        obj['page_path'] = image_path
+
+        # the processing looks a resized-padded image.
+        # but the proposal step only looks at the resized image
         pad_img_path = f'{image_path}_pad'
-        padded_img.save(pad_img_path, "PNG")
-        infofiles[pad_img_path] = 'pad'
+
+        proposals = obj.get('proposals')
+        make_proposals = proposals is None or just_propose
+
+        # if we already have proposals, then we don't need to load the non-padded image
+        # we can just load the padded image.  If there is no padded image then we 
+        # load the page image and resize/pad it.
+        if os.path.isfile(pad_img_path) and not make_proposals:
+            padded_img = load_image(pad_img_path)
+            if padded_img is None:
+                tlog_flush(f'ERROR: failed to read padded image {pad_img_path} - aborting this pdf')
+                return (False, objs, infofiles)
+
+            orig_w = obj['orig_w']
+            orig_h = obj['orig_h']
+        else:
+            # get the page image
+            img = load_image(image_path)
+            if img is None:
+                tlog(f'ERROR: failed to read image {image_path} - aborting this pdf')
+                return (False, objs, infofiles)
+
+            tlog(f'{image_name} is {img.size} resizing to 1920')
+            orig_w, orig_h = img.size
+            obj['orig_w'] = orig_w
+            obj['orig_h'] = orig_h
+
+            # the model wants square images of size 1920, so we resize and square it now
+            img, img_size = resize_png(img, return_size=True)
+            tlog(f'{page_name} is now {img_size}')
+            padded_img = pad_image(img)
+            tlog(f'{page_name} padded to {padded_img.size}')
+            # the padding operation is
+            #want_image_size = 1920
+            #d_w = want_image_size - w
+            #d_h = want_image_size - h
+            #padding = (0,0,d_w, d_h)
+            #padded_img = ImageOps.expand(img, padding, fill="#fff")
+
+            # the aggregation code also needs access to the padded image
+            # so save this file for later
+            padded_img.save(pad_img_path, "PNG")
+            infofiles[pad_img_path] = 'pad'
+
+        # refresh the path to the padded image in case we are resuming on a different machine
         obj['pad_img'] = pad_img_path
 
+        # if meta/limit is supplied, store/refresh the page relevent meta info
         if limit is not None:
            obj['pdf_limit'] = limit
```

Comment:
NIT: Just noticed that this line has sliiightly different spacing than others at the same level.